src/utils/user/userManagement.ts

```diff
@@ -1,110 +1,167 @@
- * Check if we need to update user external attributes and perform it when needed.
+ * Check if we need to update user groups and attributes and perform it when needed.
  *
- * @param setting Global setting of the platform.
  * @param user Logged user to update.
  * @param req Original req.
  * @returns Boolean to indicate if there is any change in the user.
  */
-export const updateUserAttributes = async (
-  setting: Setting,
-  user: User,
-  req: any
-): Promise<boolean> => {
+export const updateUser = async (user: User, req: any): Promise<boolean> => {
   // Check if we really need to fetch new ones
-  if (!userNeedsExternalAttributes(setting, user)) return false;
-  // Get delegated token
-  const upstreamToken = req.headers.authorization.split(' ')[1];
-  const token = await getDelegatedToken(
-    setting.userManagement.apiConfiguration,
-    user.id,
-    upstreamToken
-  );
-  let res;
-  // Fetch new attributes
-  try {
-    res = await fetch(setting.userManagement.serviceAPI, {
-      method: 'get',
-      headers: {
-        Authorization: `Bearer ${token}`,
-      },
-    });
-  } catch (e) {
-    logger.error(i18next.t('errors.invalidAPI'));
-    return false;
+  if (!userNeedsUpdate(user)) return false;
+  const userChanges: boolean[] = await Promise.all([
+    updateUserAttributes(user, req),
+    updateUserGroups(user, req),
+  ]);
+  for (const update of userChanges) {
+    if (update) return true;
   }
-  let json: any;
-  try {
-    json = await res.json();
-  } catch (e) {
-    logger.error(i18next.t('errors.authenticationTokenNotFound'));
-    return false;
+  return false;
+};
+
+/**
+ * Callback executed after authentication of user..
+ * Auto-assigned roles are added at this stage.
+ *
+ * @param error process error
+ * @param done done callback of authentication process
+ * @param token current token
+ * @param user current user
+ * @returns authentication process callback
+ */
+export const userAuthCallback = async (
+  error: any,
+  done: any,
+  token: any,
+  user: User
+) => {
+  if (error) {
+    return done(error);
   }
-  // Map them to user attributes
-  for (const mapping of setting.userManagement.attributesMapping) {
-    const attribute = get(json, mapping.path);
-    const value = get(attribute, mapping.value);
-    const text = get(attribute, mapping.text);
-    if (!mapping.field.includes('attributes') && !mapping.field.includes('.')) {
-      set(user, mapping.field, value);
-    } else {
-      set(user, mapping.field, { value, text });
-    }
+  const cacheKey = user.id + ROLES_KEY;
+  const cacheValue: any[] = cache.get(cacheKey);
+  if (!isNil(cacheValue)) {
+    console.log('++++++++++');
+    console.log(cacheValue);
+    // const autoAssignedRoles = cacheValue as Role[];
+    console.log(`Assigned roles are : ${JSON.stringify(cacheValue)}`);
+    console.log(`Default roles are : ${user.roles}`);
+    // user.roles = [...cloneDeep(cacheValue)];
+    user.roles = [...user.roles, ...cacheValue];
```

Comment:
@PacomeRivier issue I see:
- if we just use the getAutoAssignedRoles, that works fine, and the roles are correctly used
- if we store them with the cache, the permissions are broken. In fact, it only saves the permissions IDs ( so each object in the array we get from the cache only contains the array of permissions, not the permissions themselves )

You can replicate by using the keycloak config, setting a group, creating a auto assign rule for a role, then connect with an user part of the group, and no role, so the auto assigned role should be assigned to this user. Then, try to navigate between back-office pages and check the logs